File: Python/Program/Garage.py
'''
Saya Nabil Hanif Achmaddiredja mengerjakan Latihan Praktikum 4 dalam 
mata kuliah Desain dan Pemrograman Berorientasi Objek untuk 
keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin.
'''

class Garage:
    
    #Private Attributes
    __nama = ""
    __luas = ""
    __kapasitas = 0
    __vehicleList = []
    __currentCount = 0
    
    # Constructor
    def __init__(self, nama="", luas="", kapasitas=0, vehicleList=[], currentCount=0):
        self.__nama = nama
        self.__luas = luas
        self.__kapasitas = kapasitas
        self.__vehicleList = vehicleList
        self.__currentCount = currentCount
        
    # Getter and Setter
    # Getter 
    def getNama(self):
        return self.__nama

    def getLuas(self):
        return self.__luas
    
    def getKapasitas(self):
        return self.__kapasitas

    def getVehicleList(self):
        return self.__vehicleList
    
    def getCurrentCount(self):
        return self.__currentCount
    
    # Setter
    def setNama(self, nama):
        self.__nama = nama
    
    def setLuas(self, luas):
        self.__luas = luas

    def setKapasitas(self, kapasitas):
        self.__kapasitas = kapasitas

    def setVehicleList(self, vehicleList):
        self.__vehicleList = vehicleList
   
    def setCurrentCount(self, currentCount):
        self.__currentCount = currentCount

File: Cpp/Program/Garage.cpp
/*Saya Nabil Hanif Achmaddiredja mengerjakan Latihan Praktikum 4 dalam 
mata kuliah Desain dan Pemrograman Berorientasi Objek untuk 
keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin.*/

#include <iostream>
#include <string>
#include <vector>

using namespace std;

class Garage 
{
    /*Attribute*/
    private:
        string nama;
        string luas;
        int kapasitas;
        vector<string> vehicleList;
        int currentCount;

    public:
        /*Constructor*/
        Garage(){

        }

        Garage(string nama, string luas, int kapasitas) {
            this->nama = nama;
            this->luas = luas;
            this->kapasitas = kapasitas;
        }

        Garage(string nama, string luas, int kapasitas, vector<string> vehicleList, int currentCount) {
            this->nama = nama;
            this->luas = luas;
            this->kapasitas = kapasitas;
            this->vehicleList = vehicleList;
            this->currentCount = currentCount;
        }

        /*Getter & Setter*/
        //Getter
        string getNama() {
            return this->nama;
        }

        string getLuas() {
            return this->luas;
        }

        int getKapasitas() {
            return this->kapasitas;
        }

        vector<string> getVehicleList() {
            return this->vehicleList;
        }

        int getCurrentCount() {
            return this->currentCount;
        }

        //Setter
        void setNama(string nama) {
            this->nama = nama;
        }

        void setLuas(string luas) {
            this->luas = luas;
        }

        void setKapasitas(int kapasitas) {
            this->kapasitas = kapasitas;
        }

        void setVehicleList(vector<string> vehicleList) {
            this->vehicleList = vehicleList;
        }

        void setCurrentCount(int currentCount) {
            this->currentCount = currentCount;
        }
    ~Garage()
    {}
};
